handler/src/main/java/io/netty/handler/traffic/TrafficCounter.java

```diff
@@ -263,6 +237,31 @@ synchronized void resetAccounting(long newLastTime) {
      * Constructor with the {@link AbstractTrafficShapingHandler} that hosts it, the Timer to use, its
      * name, the checkInterval between two computations in millisecond.
      *
+     * @param executor
+     *            the underlying executor service for scheduling checks, might be null when used
+     * from {@link GlobalChannelTrafficCounter}.
+     * @param name
+     *            the name given to this monitor.
+     * @param checkInterval
+     *            the checkInterval in millisecond between two computations.
+     */
+    public TrafficCounter(ScheduledExecutorService executor, String name, long checkInterval) {
+        trafficShapingHandler = null;
```

Comment:
Maybe you were a little lazy? :stuck_out_tongue: 
Should add a tiny helper for use by both the constructors?

```java
private void init(AbstractTrafficShapingHandler trafficShapingHandler, ScheduledExecutorService executor,
        String name, long checkInterval) {
    this.trafficShapingHandler = trafficShapingHandler;
    this.executor = executor;
    this.name = name;
    // absolute time: informative only
    lastCumulativeTime = System.currentTimeMillis();
    writingTime = milliSecondFromNano();
    readingTime = writingTime;
    lastWritingTime = writingTime;
    lastReadingTime = writingTime;
    configure(checkInterval);
}
```